src/xrc/src/forex/nepal.rs

```diff
@@ -0,0 +1,164 @@
+    #[serde(skip_deserializing)]
+    #[serde(rename(deserialize = "errors"))]
+    _errors: String,
+    #[serde(skip_deserializing)]
+    #[serde(rename(deserialize = "params"))]
+    _params: String,
+    data: CentralBankOfNepalData,
+    #[serde(skip_deserializing)]
+    #[serde(rename(deserialize = "pagination"))]
+    _pagination: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct CentralBankOfNepalStatus {
+    code: u64,
+}
+
+#[derive(Debug, Deserialize)]
+struct CentralBankOfNepalData {
+    payload: Vec<CentralBankOfNepalDataOneDay>,
+}
+
+#[derive(Debug, Deserialize)]
+struct CentralBankOfNepalDataOneDay {
+    date: String,
+    #[serde(rename(deserialize = "published_on"))]
+    _published_on: String,
+    #[serde(rename(deserialize = "modified_on"))]
+    _modified_on: String,
+    rates: Vec<CentralBankOfNepalDataRate>,
+}
+
+#[derive(Debug, Deserialize)]
+struct CentralBankOfNepalDataRate {
+    currency: CentralBankOfNepalCurrency,
+    buy: String,
+    sell: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct CentralBankOfNepalCurrency {
+    iso3: String,
+    #[serde(rename(deserialize = "name"))]
+    _name: String,
+    unit: u64,
+}
+
+impl IsForex for CentralBankOfNepal {
+    fn format_timestamp(&self, timestamp: u64) -> String {
+        format!(
+            "{}",
+            NaiveDateTime::from_timestamp(timestamp.try_into().unwrap_or(0), 0).format("%Y-%m-%d")
+        )
+    }
+
+    fn extract_rate(&self, bytes: &[u8], timestamp: u64) -> Result<ForexRateMap, ExtractError> {
+        let response = serde_json::from_slice::<CentralBankOfNepalStruct>(bytes)
+            .map_err(|err| ExtractError::json_deserialize(bytes, err.to_string()))?;
+
+        if response.status.code != 200 {
+            return Err(ExtractError::RateNotFound {
+                filter: format!(
+                    "Cannot find data for timestamp (status code: {})",
+                    response.status.code
+                ),
+            });
+        }
+
+        let timestamp = (timestamp / SECONDS_PER_DAY) * SECONDS_PER_DAY;
+        let mut values = ForexRateMap::new();
+
+        for day in response.data.payload {
+            let extracted_timestamp =
+                NaiveDateTime::parse_from_str(&(day.date + " 00:00:00"), "%Y-%m-%d %H:%M:%S")
+                    .unwrap_or_else(|_| NaiveDateTime::from_timestamp(0, 0))
+                    .timestamp() as u64;
+            if extracted_timestamp == timestamp {
+                for rate in day.rates {
+                    let buy = rate.buy.parse::<f64>();
+                    let sell = rate.sell.parse::<f64>();
+                    let value = match (buy, sell) {
+                        (Ok(buy), Ok(sell)) => ((buy + sell) / 2.0 * (RATE_UNIT as f64)) as u64,
+                        _ => continue,
+                    };
+                    values.insert(rate.currency.iso3, value / rate.currency.unit);
+                }
+                break;
+            }
+        }
```

Comment:
A slightly different way to do it that avoids the need for `break` and `continue`.
```suggestion
        let day = response
            .data
            .payload
            .iter()
            .find(|day| {
                let date = format!("{} 00:00:00", day.date);
                let extracted_timestamp = NaiveDateTime::parse_from_str(&date, "%Y-%m-%d %H:%M:%S")
                    .unwrap_or_else(|_| NaiveDateTime::from_timestamp(0, 0))
                    .timestamp() as u64;
                extracted_timestamp == timestamp
            })
            .ok_or_else(|| ExtractError::RateNotFound {
                filter: "Cannot find data for timestamp".to_string(),
            })?;

        let values = day
            .rates
            .iter()
            .filter_map(|rate| {
                let buy = rate.buy.parse::<f64>();
                let sell = rate.sell.parse::<f64>();
                let value = match (buy, sell) {
                    (Ok(buy), Ok(sell)) => ((buy + sell) / 2.0 * (RATE_UNIT as f64)) as u64,
                    _ => return None,
                };
                Some((rate.currency.iso3.clone(), value / rate.currency.unit))
            })
            .collect::<ForexRateMap>();
```